### PR TITLE
feat(install): Add help argument to install.sh

### LINKED ISSUE #2727: Starship won't install under Solus 4.2
#### Current Behavior
Starship won't install.

#### Expected Behavior
Install Starship.

#### Additional context/Screenshots
```
sh -c "$(curl -fsSL https://starship.rs/install.sh)"

  Configuration
> Bin directory: /usr/local/bin
> Platform:      unknown-linux-musl
> Arch:          x86_64

> Tarball URL: https://github.com/starship/starship/releases/latest/download/starship-x86_64-unknown-linux-musl.tar.gz
? Install Starship latest to /usr/local/bin? [y/N] y
x Installation location /usr/local/bin does not appear to be a directory
```

#### Environment
- Shell type: bash
- Shell version: GNU bash, version 5.1.8(1)-release (x86_64-solus-linux-gnu)
- Terminal emulator: kitty
- Operating system: Solus 4.2 Fortitude

### PR DESCRIPTION
**Pull request by andytom**
#### Description

Have added a basic help to the install script to make it a bit easier to
use.

#### Motivation and Context

Closes #2727

#### How Has This Been Tested?

- [ ] I have tested using **MacOS**
- [x] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -161,6 +161,13 @@ shown below. Can't see yours? Have a look at the [extra platform instructions](h
    ```
    To update the Starship itself, rerun the above script. It will replace the current version without touching Starship's configuration.
 
+
+   **Note** - The defaults of the install script can be overridden see the build in help.
```
Comment (kidonng) [inline README.md:165]:
```suggestion
   **Note** - The defaults of the install script can be overridden see the build in help.
  **Note** - The defaults of the install script can be overridden see the built-in help.
```

**Review by kidonng — COMMENTED** [review]
Looking good! Only spotted a typo.

**Review by kidonng — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (andytom) [reply README.md:165]:
```suggestion 
```

File: README.md [hunk 2]
```diff
@@ -161,6 +161,14 @@ shown below. Can't see yours? Have a look at the [extra platform instructions](h
...
+  **Note** - The defaults of the install script can be overridden see the built-in help.
```
Comment (davidkna) [inline README.md:166]: The line is duplicated and the old indentation was correct:
```suggestion
   **Note** - The defaults of the install script can be overridden see the built-in help.
```

**Review by vladimyr — APPROVED** [review]
LGTM

File: README.md [hunk 3]
```diff
@@ -163,6 +163,7 @@ shown below. Can't see yours? Have a look at the [extra platform instructions](h
 
 
    **Note** - The defaults of the install script can be overridden see the build in help.
```
Comment (matchai) [inline README.md]: It looks like we still have this lingering line 😄 
```suggestion
```

File: README.md (hunk 2 above)
Comment (kidonng) [reply README.md]: 😅 look like I mistook `suggestion` block as `diff`

File: README.md (hunk 3 above)
Comment (kidonng) [reply README.md]: Isn't that already fixed 👀

**Pull request merged**